### PR TITLE
Fix num genes vs num umis

### PR DESCRIPTION
**Pull request by seb-mueller**
# Background
#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

- fixed qc-runner/src/numGenesVsNumUmis.r which failed when disabling this filter
- added traceback option to allow for more extensive debugging

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ npm start
 
 ## Debugging locally
 
+
```
Comment (seb-mueller) [inline README.md:59]:
```suggestion

```

File: qc-runner/src/init.r [hunk 2]
```diff
@@ -98,7 +98,9 @@ run_step <- function(scdata, config, task_name, sample_id, debug_config) {
 
 
 handle_debug <- function(scdata, config, task_name, sample_id, debug_config) {
+
     is_debug <- debug_config$step %in% c(task_name, 'all')
+
```
Comment (seb-mueller) [inline qc-runner/src/init.r:103]:
```suggestion
    is_debug <- debug_config$step %in% c(task_name, 'all')
```

**Review by seb-mueller — COMMENTED** [review]
del empty lines

File: qc-runner/src/numGenesVsNumUmis.r [hunk 3]
```diff
@@ -84,27 +85,30 @@ task <- function(seurat_obj, config, task_name, sample_id, num_cells_to_downsamp
             #Because we have subseted the object before creating the dataframes, and just selected the outlier values from that df
             #In the end we wil only remove values that are in the desired sample. 
             seurat_obj.filtered <- subset_safe(seurat_obj,colnames(seurat_obj)[!colnames(seurat_obj)%in%outliers])
+                #Similarly, when creating the plot from the df, we will just be using the values for the required sample.
+            plot1_data <- unname(purrr::map2(df$molecules,df$genes,function(x,y){c("log_molecules"=x,"log_genes"=y)}))
```
Comment (alexvpickering) [inline qc-runner/src/numGenesVsNumUmis.r:89]:
```suggestion
            #Similarly, when creating the plot from the df, we will just be using the values for the required sample.
            plot1_data <- unname(purrr::map2(df$molecules,df$genes,function(x,y){c("log_molecules"=x,"log_genes"=y)}))
```

**Review by alexvpickering — APPROVED** [review]
This looks good to me. My understanding is that we are moving the plot generation to happen only if it is enabled for the sample whereas before it disregarded the flag (and would throw an error because df doesn't exist).

 Please check the suggestions

**Review by ogibson-biomage — APPROVED** [review]

**Pull request merged**